rasa/utils/tensorflow/layers.py

```diff
@@ -121,6 +126,33 @@ def __init__(self, reg_lambda: float = 0, **kwargs: Any) -> None:
 
         super().__init__(kernel_regularizer=regularizer, **kwargs)
 
+    def get_units(self) -> int:
+        """Returns number of output units."""
+        return self.units
+
+    def get_kernel(self) -> tf.Tensor:
+        """Returns kernel tensor."""
+        return self.kernel
+
+    def get_bias_info(self) -> Tuple[bool, tf.Tensor]:
+        """Returns whether the layer has a bias tensor and the bias tensor itself."""
+        return self.use_bias, self.bias
+
+    def get_feature_type(self) -> Union[Text, None]:
+        """Returns a feature type of the data that's fed to the layer."""
+        for feature_type in [FEATURE_TYPE_SENTENCE, FEATURE_TYPE_SEQUENCE]:
+            if feature_type in self.name:
+                return feature_type
+        return None
+
+    def get_attribute(self) -> Union[Text, None]:
+        """Returns an attribute of the data that's fed to the layer."""
```

Comment:
```suggestion
        """Returns the attribute for which this layer was constructed. For example: TEXT."""
```

Reply:
Returns section missing in docstring.

Reply:
Also, please describe the format you expect the name to be in for this method to actually work.

Reply:
Also, you should return the attribute only if it is one of `[TEXT, INTENT, LABEL, DIALOGUE, ACTION_NAME, ACTION_TEXT, f"{LABEL}_{ACTION_NAME}", f"{LABEL}_{ACTION_TEXT}"]`. You can find these constants imported inside `rasa/core/policies/ted_policy.py`.
This will ensure that we only return valid attribute types and prevents cases when the layer is instantiated with a name `sparse_to_dense.xyz_sentence`.

Reply:
then why don't we just iterate over the possible attributes and see if they're included in the name?